src/annotator/guest.js

```diff
@@ -293,25 +291,24 @@ export default class Guest {
     });
 
     this.crossframe.on('scrollToAnnotation', tag => {
-      for (let anchor of this.anchors) {
-        if (anchor.highlights) {
-          if (anchor.annotation.$tag === tag) {
-            const range = resolveAnchor(anchor);
-            if (!range) {
-              continue;
-            }
-
-            const event = new CustomEvent('scrolltorange', {
-              bubbles: true,
-              cancelable: true,
-              detail: range,
-            });
-            const defaultNotPrevented = this.element.dispatchEvent(event);
-            if (defaultNotPrevented) {
-              scrollIntoView(anchor.highlights[0]);
-            }
-          }
-        }
+      const anchor = this.anchors.find(a => a.annotation.$tag === tag);
+      if (!anchor?.highlights) {
+        return;
+      }
+      const range = resolveAnchor(anchor);
+      if (!range) {
+        return;
+      }
+
+      const event = new CustomEvent('scrolltorange', {
```

Comment:
It's not immediately clear what the intention of this event is, and its following default-prevented check. It is to allow something to supersede the scrolling behavior? Can some commenting or clarity be added here?

Reply:
The event was added for the benefit of third parties IIRC. The event name is referenced in the `docs/developers/arch/adr-002.md` doc. I'm not sure if this is actively in use, but I left it as-is here.

Reply:
From re-reading that doc, it looks like the original use case for `scrolltorange` was similar to the [`beforematch` event](https://www.chromestatus.com/feature/5400510406328320) which Chrome introduced recently. We might want to adopt that event in future so that we can benefit from more pages supporting it.